button.lua

```diff
@@ -28,17 +33,24 @@ function PrimeUI.button(win, x, y, text, action, fgColor, bgColor, clickedColor)
     -- Get the screen position and add a click handler.
     PrimeUI.addTask(function()
         local buttonDown = false
+        local timer = nil
         while true do
             local event, button, clickX, clickY = os.pullEvent()
             local screenX, screenY = PrimeUI.getWindowPos(win, x, y)
-            if event == "mouse_click" and button == 1 and clickX >= screenX and clickX < screenX + #text + 2 and clickY == screenY then
+            if (not monitor and event == "mouse_click" and button == 1 and clickX >= screenX and clickX < screenX + #text + 2 and clickY == screenY) or
+                (event == "monitor_touch" and clickX >= screenX and clickX < screenX + #text + 2 and clickY == screenY)
+            then
                 -- Initiate a click action (but don't trigger until mouse up).
                 buttonDown = true
                 -- Redraw the button with the clicked background color.
                 win.setCursorPos(x, y)
                 win.setBackgroundColor(clickedColor)
                 win.setTextColor(fgColor)
                 win.write(" " .. text .. " ")
+
+                if monitor then
```

Comment:
As above, comment about why we need a timer here.

Reply:
Done.